lib/study.py

```diff
@@ -0,0 +1,248 @@
+        madx_sec['madx_exe'] = self.paths['madx_exe']
+        madx_sec['mask_name'] = self.madx_input["mask_name"]
+        madx_sec['output_files'] = utils.code(self.madx_output)
+
+        self.config['mask'] = {}
+        mask_sec = self.config['mask']
+
+        self.config['sixtrack'] = {}
+        six_sec = self.config['sixtrack']
+        six_sec['source_path'] = self.paths['location']
+        six_sec['sixtrack_exe'] = self.paths['sixtrack_exe']
+        six_sec['temp_files'] = utils.code(self.oneturn_sixtrack_input['temp'])
+        six_sec['input_files'] = utils.code(self.oneturn_sixtrack_input['input'])
+        six_sec['output_files'] = utils.code(self.oneturn_sixtrack_output)
+        self.config['fort3'] = self.oneturn_sixtrack_params
+
+        keys = sorted(self.madx_params.keys())
+        values = []
+        for key in keys:
+            values.append(self.madx_params[key])
+
+        for element in itertools.product(*values):
+            for i in range(len(element)):
+                ky = keys[i]
+                vl = element[i]
+                mask_sec[ky] = str(vl)
+            input_name = self.madx_name_conven('mad6t', keys, element, '.ini')
+            prefix = self.madx_input['mask_name'].split('.')[0]
+            madx_input_name = self.madx_name_conven(prefix, keys, element)
+            madx_sec['input_name'] = madx_input_name
+            out_name = self.madx_name_conven('Result', keys, element, '')
+            mad6t_input = os.path.join(self.location, 'mad6t_input')
+            madx_sec['dest_path'] = os.path.join(self.paths['madx_out'], out_name)
+            six_sec['dest_path'] = os.path.join(self.paths['madx_out'], out_name)
+            output = os.path.join(mad6t_input, input_name)
+            with open(output, 'w') as f_out:
+                self.config.write(f_out)
+            self.mad6t_joblist.append(output)
+            print('Successfully generate input file %s'%output)
+
+    def madx_name_conven(self, prefix, keys, values, suffix = '.madx'):
+        '''The convention for naming input file'''
+        mk = prefix
+        num = len(keys) if len(keys)<len(values) else len(values)
+        for i in range(num):
+            mk += '_%s_%s'%(keys[i], str(values[i]))
+        mk += suffix
+        return mk
+
+class StudyFactory(object):
+
+    def __init__(self, workspace='./sandbox'):
+        self.ws = workspace
+        self.studies = []
+        self._setup_ws()
+
+    def _setup_ws(self):
+        '''Setup a workspace'''
+        if not os.path.isdir(self.ws):
+            os.mkdir(self.ws)
+            print('Create new workspace %s!'%self.ws)
+        else:
+            print('The workspace %s already exists!'%self.ws)
+        studies = os.path.join(self.ws, 'studies')
+        if not os.path.isdir(studies):
+            os.mkdir(studies)
+
+    def info(self):
+        '''Print all the studies in the current workspace'''
+        return self.studies
```

Comment:
this function doesn't really print... :D

Reply:
yes....  inappropriate comments. I will add a ```print``` statement here.